internal/zoom/client_test.go

```diff
@@ -0,0 +1,133 @@
+package zoom
+
+import (
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type TestCase struct {
+	MeetingID string
+	Error     error
+	Result    []Participant
+}
+
+func dummyHandler() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		var message []byte
+
+		meetingID := strings.Split(r.URL.Path, "/")[3]
+
+		switch meetingID {
+		case "error":
+			w.WriteHeader(http.StatusInternalServerError)
+			message = []byte(`{"code": 500, "message": "invalid request"}`)
+		default:
+			message = []byte(`
+			{
+				"page_count": 1,
+				"page_size": 30,
+				"total_records": 2,
+				"next_page_token": "",
+				"participants": [
+					{
+						"id": "d52f19c548b88490b5d16fcbd38",
+						"user_id": "32dsfsd4g5gd",
+						"user_name": "dojo",
+						"device": "WIN",
+						"ip_address": "127.0.0.1",
+						"location": "New York",
+						"network_type": "Wired",
+						"microphone": "Plantronics BT600",
+						"camera": "FaceTime HD Camera",
+						"speaker": "Plantronics BT600",
+						"data_center": "SC",
+						"connection_type": "P2P",
+						"join_time": "2019-09-07T13:15:02.837Z",
+						"leave_time": "2019-09-07T13:15:09.837Z",
+						"share_application": false,
+						"share_desktop": true,
+						"share_whiteboard": true,
+						"recording": false,
+						"status": "in_waiting_room",
+						"pc_name": "dojo's pc",
+						"domain": "Dojo-workspace",
+						"mac_addr": " 00:0a:95:9d:68:16",
+						"harddisk_id": "sed proident in",
+						"version": "4.4.55383.0716",
+						"leave_reason": "Dojo left the meeting.<br>Reason: Host ended the meeting."
+					},
+					{
+						"id": "z8aaaaaaCfp8uQ",
+						"user_id": "1670000000",
+						"user_name": "Rea",
+						"device": "Android",
+						"ip_address": "120.000.000",
+						"location": "San Jose (US)",
+						"network_type": "Wifi",
+						"data_center": "SC",
+						"connection_type": "UDP",
+						"join_time": "2019-08-02T15:31:48Z",
+						"leave_time": "2019-08-02T16:04:12Z",
+						"share_application": false,
+						"share_desktop": false,
+						"share_whiteboard": false,
+						"recording": false,
+						"pc_name": "Rea's PC",
+						"domain": "Rea-workspace",
+						"mac_addr": "",
+						"harddisk_id": "",
+						"version": "4.4.55383.0716",
+						"leave_reason": "Rea left the meeting.<br>Reason: Host closed the meeting."
+					}
+				]
+			}`)
+		}
+		w.Write(message)
+	})
+}
+
+func TestGetMeetingParticipants(t *testing.T) {
+	ts := httptest.NewServer(dummyHandler())
+	token := "token"
+	secret := "secret"
+	meetingID := "meetingID"
+	client := NewClient(token, secret, meetingID, ts.URL)
+
+	testCases := []TestCase{
+		{
+			MeetingID: "error",
```

Comment:
It is better to test with real data to ensure handlers work properly. It is also important to check other expected attributes in the handler - like that the expected input is properly passed (request URL, headers, etc). Using an inline callback or factory of such callbacks helps.

Reply:
Can you give me some example please?